### PR TITLE
Removes some spawns from explosion effects

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by KorPhaeron** [comment]
This also does not need a changelog. Why would a player want or need to know this?

**Comment by MoreRobustThanYou** [comment]
@KorPhaeron why does the "code" tag exist again?

**Comment by KorPhaeron** [comment]
Maybe for big refactors or changes in coding standards or something. Something so small is just clutter though, and in this case probably confusing clutter since "spawn" as used in byond vs what the average player thinks of when they read "spawn" in a game is entirely unrelated

File: code/game/objects/effects/effect_system/effects_explosion.dm [hunk 1]
```diff
@@ -11,15 +11,16 @@
...
+		step(expl, direct)
+	
 /datum/effect_system/expl_particles/start()
 	for(var/i in 1 to number)
-		spawn(0)
-			var/obj/effect/particle_effect/expl_particles/expl = new /obj/effect/particle_effect/expl_particles(location)
-			var/direct = pick(GLOB.alldirs)
-			var/steps_amt = pick(1;25,2;50,3,4;200)
-			for(var/j in 1 to steps_amt)
-				sleep(1)
```
Comment (ChangelingRain) [inline code/game/objects/effects/effect_system/effects_explosion.dm]: where'd this sleep go

File: code/game/objects/effects/effect_system/effects_explosion.dm [hunk 2]
```diff
@@ -11,15 +11,16 @@
 /datum/effect_system/expl_particles
 	number = 10
 
+/datum/effect_system/expl_particles/proc/create_particles()
+	var/obj/effect/particle_effect/expl_particles/expl = new /obj/effect/particle_effect/expl_particles(location)
+	var/direct = pick(GLOB.alldirs)
+	var/steps_amt = pick(1;25,2;50,3,4;200)
+	for(var/j in 1 to steps_amt)
+		step(expl, direct)
+
```
Comment (Cyberboss) [inline code/game/objects/effects/effect_system/effects_explosion.dm]: Put the sleep back

**Review by Cyberboss — CHANGES_REQUESTED** [review]
Bring back the slee

File: code/game/objects/effects/effect_system/effects_explosion.dm [hunk 3]
```diff
@@ -16,8 +16,7 @@
 	var/direct = pick(GLOB.alldirs)
 	var/steps_amt = pick(1;25,2;50,3,4;200)
 	for(var/j in 1 to steps_amt)
-		sleep(1)
-		step(expl, direct)
+		addtimer(CALLBACK(src, .proc/_step, expl, direct), 1)
```
Comment (Cyberboss) [inline code/game/objects/effects/effect_system/effects_explosion.dm]: 1 -> j

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/objects/effects/effect_system/effects_explosion.dm [hunk 4]
```diff
@@ -11,15 +11,16 @@
...
 /datum/effect_system/expl_particles/start()
 	for(var/i in 1 to number)
-		spawn(0)
-			var/obj/effect/particle_effect/expl_particles/expl = new /obj/effect/particle_effect/expl_particles(location)
-			var/direct = pick(GLOB.alldirs)
-			var/steps_amt = pick(1;25,2;50,3,4;200)
-			for(var/j in 1 to steps_amt)
-				sleep(1)
-				step(expl,direct)
+		INVOKE_ASYNC(src, .proc/create_particles)
```
Comment (Cyberboss) [inline code/game/objects/effects/effect_system/effects_explosion.dm]: This is no longer needed

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/objects/effects/effect_system/effects_explosion.dm (hunk 4 above)
Comment (Cyberboss) [reply code/game/objects/effects/effect_system/effects_explosion.dm]: The INVOKE_ASYNC

File: code/game/objects/effects/effect_system/effects_explosion.dm [hunk 5]
```diff
@@ -10,16 +10,14 @@
 
 /datum/effect_system/expl_particles
 	number = 10
-
+
```
Comment (ShizCalev) [inline code/game/objects/effects/effect_system/effects_explosion.dm]: dirtyyyyyyy

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**